libvcx/src/utils/devsetup_agent.rs

```diff
@@ -0,0 +1,462 @@
+            assert_eq!(expected_status, proof::get_proof_state(self.presentation_handle).unwrap());
+        }
+
+        pub fn teardown(&mut self) {
+            self.activate().unwrap();
+            close_main_wallet().unwrap();
+            delete_wallet(&self.config_wallet).unwrap();
+        }
+    }
+
+    pub struct Alice {
+        pub is_active: bool,
+        pub config_wallet: WalletConfig,
+        pub config_agency: AgencyConfig,
+        pub connection_handle: u32,
+        pub credential_handle: u32,
+        pub presentation_handle: u32,
+    }
+
+    impl Alice {
+        pub fn setup() -> Alice {
+            settings::clear_config();
+
+            let config_wallet = WalletConfig {
+                wallet_name: format!("alice_wallet_{}", uuid::Uuid::new_v4().to_string()),
+                wallet_key: settings::DEFAULT_WALLET_KEY.into(),
+                wallet_key_derivation: settings::WALLET_KDF_RAW.into(),
+                wallet_type: None,
+                storage_config: None,
+                storage_credentials: None,
+                rekey: None,
+                rekey_derivation_method: None
+            };
+
+            let config_provision_agent = ProvisionAgentConfig {
+                agency_did: C_AGENCY_DID.to_string(),
+                agency_verkey: C_AGENCY_VERKEY.to_string(),
+                agency_endpoint: C_AGENCY_ENDPOINT.to_string(),
+                agent_seed: None
+            };
+
+            create_wallet(&config_wallet).unwrap();
+            open_as_main_wallet(&config_wallet).unwrap();
+            let config_agency = provision_cloud_agent(&config_provision_agent).unwrap();
+            close_main_wallet().unwrap();
+
+            Alice {
+                is_active: false,
+                config_wallet,
+                config_agency,
+                connection_handle: 0,
+                credential_handle: 0,
+                presentation_handle: 0,
+            }
+        }
+
+        pub fn accept_invite(&mut self, invite: &str) {
+            self.activate().unwrap();
+            self.connection_handle = connection::create_connection_with_invite("faber", invite).unwrap();
+            connection::connect(self.connection_handle).unwrap();
+            connection::update_state(self.connection_handle).unwrap();
+            assert_eq!(3, connection::get_state(self.connection_handle));
+        }
+
+        pub fn update_state(&mut self, expected_state: u32) {
+            self.activate().unwrap();
+            connection::update_state(self.connection_handle).unwrap();
+            assert_eq!(expected_state, connection::get_state(self.connection_handle));
+        }
+
+        pub fn download_message(&mut self, message_type: PayloadKinds) -> VcxAgencyMessage {
+            self.activate().unwrap();
+            let did = connection::get_pw_did(self.connection_handle).unwrap();
+            download_message(did, message_type) // tood: need to pass PayloadKind
```

Comment:
1.) "todo"
2.) Is the comment up-to-date? It seems we are passing a PayloadKind?

Reply:
yeah this is definitely out of date, deleted